nixos/modules/services/networking/icecream/daemon.nix

```diff
@@ -0,0 +1,165 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.icecream.daemon;
+  user = "icecc";
+  group = "icecc";
+in {
+
+  ###### interface
+
+  options = {
+
+    services.icecream.daemon = {
+     enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to run the Icecream daemon.
+        '';
+      };
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to automatically open the specified ports in the firewall.
+        '';
+      };
+
+      openBroadcast = mkOption {
```

Comment:
We have an official policy against opening firewall workout the explicit consent of the user, except for `ssh`. Even in the case that a service is useless without. You could either not provide a default value forcing the user to do so, or add a `warning` if the user hasn't set the value to `true`.